src/libraries/System.Private.CoreLib/generators/EventSourceGenerator.Parser.cs

```diff
@@ -0,0 +1,339 @@
+
+                                            switch (argName)
+                                            {
+                                                case "Guid":
+                                                    guid = value;
+                                                    break;
+                                                case "Name":
+                                                    name = value;
+                                                    break;
+                                            }
+                                        }
+                                    }
+
+                                    if (!Guid.TryParse(guid, out Guid result))
+                                    {
+                                        result = GenerateGuidFromName(name.ToUpperInvariant());
+                                    }
+
+                                    eventSourceClass = new EventSourceClass
+                                    {
+                                        Namespace = nspace,
+                                        ClassName = className,
+                                        SourceName = name,
+                                        Guid = result
+                                    };
+                                    continue;
+                                }
+                            }
+                        }
+
+                        if (!autoGenerate)
+                        {
+                            continue;
+                        }
+
+                        if (eventSourceClass is null)
+                        {
+                            continue;
+                        }
+
+                        results ??= new List<EventSourceClass>();
+                        results.Add(eventSourceClass);
+                    }
+                }
+
+                return results?.ToArray() ?? Array.Empty<EventSourceClass>();
+            }
+
+            // From System.Private.CoreLib
+            private static Guid GenerateGuidFromName(string name)
+            {
+                ReadOnlySpan<byte> namespaceBytes = new byte[] // rely on C# compiler optimization to remove byte[] allocation
+                {
+                0x48, 0x2C, 0x2D, 0xB2, 0xC3, 0x90, 0x47, 0xC8,
+                0x87, 0xF8, 0x1A, 0x15, 0xBF, 0xC1, 0x30, 0xFB,
+                };
+
+                byte[] bytes = Encoding.BigEndianUnicode.GetBytes(name);
+                Sha1ForNonSecretPurposes hash = default;
+                hash.Start();
+                hash.Append(namespaceBytes);
+                hash.Append(bytes);
+                Array.Resize(ref bytes, 16);
+                hash.Finish(bytes);
+
+                bytes[7] = unchecked((byte)((bytes[7] & 0x0F) | 0x50));    // Set high 4 bits of octet 7 to 5, as per RFC 4122
+                return new Guid(bytes);
+            }
+
+            private struct Sha1ForNonSecretPurposes
```

Comment:
You can just use actual Crypto Sha1 here. No need to replicate `Sha1ForNonSecretPurposes` here.

Reply:
If the apis were made public (e.g. Guid taking .ctor) it would need to use the same algo or the Guid would change?

None of the System.Private.CoreLib eventsources use this path though as they all provide a Guid in the `EventSource` attribute

Reply:
SHA1 is standardized algorithm. We can just use the implementation from System.Security.Cryptography in the source generator.

Reply:
Ahhh... because the generator runs outside corelib, whereas corelib doesn't know about  System.Security.Cryptography

Got ya

Reply:
Done